concordium-std/src/types.rs

```diff
@@ -672,6 +687,56 @@ pub type QueryAccountBalanceResult = Result<AccountBalance, QueryAccountBalanceE
 /// [`QueryContractBalanceError`] and result to [`Amount`].
 pub type QueryContractBalanceResult = Result<Amount, QueryContractBalanceError>;
 
+/// A wrapper around [`Result`] that fixes the error variant to
+/// [`QueryAccountPublicKeysError`] and result to [`AccountPublicKeys`].
+pub type QueryAccountPublicKeysResult = Result<AccountPublicKeys, QueryAccountPublicKeysError>;
+
+/// A wrapper around [`Result`] that fixes the error variant to
+/// [`CheckAccountSignatureError`] and result to [`bool`].
+pub type CheckAccountSignatureResult = Result<bool, CheckAccountSignatureError>;
+
+pub(crate) type KeyIndex = u8;
+
+#[derive(crate::Serialize, Debug, SchemaType)]
+pub(crate) enum PublicKey {
+    Ed25519(PublicKeyEd25519),
+}
+
+#[derive(crate::Serialize, Debug, SchemaType)]
```

Comment:
At least the public types must be documented here

Reply:
added